File: arrays_hashing/valid_anagram/valid_anagram.go
package validanagram

func ValidAnagram(s, t string) bool {
	if len(s) != len(t) {
		return false
	}

	charCount := map[byte]int{}
	for i := 0; i < len(s); i++ {
		charCount[s[i]]++
		charCount[t[i]]--
	}
	for _, value := range charCount {
		if value != 0 {
			return false
		}
	}
	return true
}

File: arrays_hashing/valid_anagram/valid_anagram.py
def valid_anagram(s, t):
    if len(s) != len(t):
        return False

    count = {}
    for char in s:
        count[char] = count.get(char, 0) + 1

    for char in t:
        # If char is not in the count of S string or if the count is already
        # zero for this char (too many instances of char in T compared to S).
        if char not in count or count[char] == 0:
            return False
        count[char] -= 1
    return True
